packages/react-server/src/ReactServerStreamConfigBrowser.js

```diff
@@ -13,9 +13,18 @@ export type PrecomputedChunk = Uint8Array;
 export opaque type Chunk = Uint8Array;
 export type BinaryChunk = Uint8Array;
 
-export function scheduleWork(callback: () => void) {
-  callback();
-}
+const LocalMessageChannel =
+  typeof MessageChannel === 'function' ? MessageChannel : undefined;
+export const scheduleWork: (callback: () => void) => void =
+  LocalMessageChannel !== undefined
+    ? (callback: () => void) => {
+        const channel = new LocalMessageChannel();
```

Comment:
MessageChannel support goes back quite far. I wonder if we should just assume it

Reply:
Yea probably fine to assume and maybe encourages not abusing this build for other environments. Although we currently abuse it by using it in Deno. https://github.com/facebook/react/blob/main/packages/react-dom/package.json#L64

You don't need to create a new MessageChannel each time. Can use the same one and just post a new message to it like we do in the scheduler.

Reply:
the only safe way I think we can do that is to use a queue